src/Tools/dotnet-monitor/Auth/AzureAd/AzureAdAuthConfigurator.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Diagnostics.Tools.Monitor.Swagger;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Identity.Web;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerGen;
+using Swashbuckle.AspNetCore.SwaggerUI;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Auth.AzureAd
+{
+    internal sealed class AzureAdAuthConfigurator : IAuthenticationConfigurator
+    {
+        private readonly AzureAdOptions _azureAdOptions;
+        private readonly string _fqSwaggerScope;
+
+        public AzureAdAuthConfigurator(AzureAdOptions azureAdOptions)
+        {
+            _azureAdOptions = azureAdOptions;
+
+            if (_azureAdOptions.SwaggerScope != null)
+            {
+                _fqSwaggerScope = new Uri(_azureAdOptions.GetAppIdUri(), _azureAdOptions.SwaggerScope).ToString();
+            }
+        }
+
+        public void ConfigureApiAuth(IServiceCollection services, HostBuilderContext context)
+        {
+            services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+                .AddMicrosoftIdentityWebApi(
+                    configureJwtBearerOptions: options =>
+                    {
+                        options.Audience = _azureAdOptions.GetAppIdUri().ToString();
+                    },
+                    configureMicrosoftIdentityOptions: options =>
+                    {
+                        options.Instance = _azureAdOptions.GetInstance().ToString();
+                        options.TenantId = _azureAdOptions.GetTenantId();
+                        options.ClientId = _azureAdOptions.ClientId;
+                    }
+                );
+
+            services.AddAuthorization(options =>
+            {
+                options.AddPolicy(AuthConstants.PolicyName, (builder) =>
```

Comment:
So to be clear, this will use the same [Authorize(Policy = AuthConstants.PolicyName)] attribute on our controllers correct?